common-rest/src/main/java/org/uniprot/api/rest/download/queue/BaseAbstractMessageListener.java

```diff
@@ -65,13 +70,18 @@ public void onMessage(Message message) {
                         "Message with job id {} discarded after max retry {}",
                         jobId,
                         getMaxRetryCount());
+                asyncDownloadFileHandler.deleteAllFiles(jobId);
                 return;
```

Comment:
I think, we should reset the counts in if block after maximum retry. You are just cleaning up the files.
I feel, you should have called cleanBeforeRetry(jobId); here too instead of asyncDownloadFileHandler.deleteAllFiles(jobdi);

If you make this change, you need to rename method cleanBeforeRetry